driver/src/contracts/stablex_contract.rs

```diff
@@ -40,11 +42,11 @@ pub struct StableXContractImpl {
 }
 
 impl StableXContractImpl {
-    pub fn new(web3: &contracts::Web3, key: PrivateKey, network_id: u64) -> Result<Self> {
+    pub async fn new(web3: &contracts::Web3, key: PrivateKey, network_id: u64) -> Result<Self> {
         let defaults = contracts::method_defaults(key, network_id)?;
 
-        let viewer = BatchExchangeViewer::deployed(&web3).wait()?;
-        let mut instance = BatchExchange::deployed(&web3).wait()?;
+        let viewer = BatchExchangeViewer::deployed(&web3).await?;
+        let mut instance = BatchExchange::deployed(&web3).await?;
```

Comment:
So I see that here we changed `wait()` to `await` but lower down we are putting `.wait()` on things? Is this intended?

Reply:
There is no `wait()` in this file. If you mean other files then it is intended everywhere where the function containing that code is not itself async. For example in tests it will stay this way and in other places we are going to push the async forward in following PRs.
If there is ever a `wait()` inside an `async fn` then that is a mistake.